homework/insensitive-palindrom/palindrome.cpp

```diff
@@ -0,0 +1,32 @@
+#include "palindrome.hpp"
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+#include <iterator>
+
+std::string clean_string(const std::string& string) {
+    std::string cleaned;
+    std::copy_if(begin(string),
+                 end(string),
+                 std::back_inserter(cleaned),
+                 [](auto sign) {
+                     return ('A' <= sign && sign <= 'Z') || ('a' <= sign && sign <= 'z');
+                 });
```

Comment:
I know this is not part of code for embedded systems etc ;p But generally better option is to copy string and use remove_if. Try to think why and then scroll down to answer ;p
.
.
.
.
.
.
.
.
.
.
.
.
.
Beacuse for long string we can do a lot of allocation (2, 4, 8, 16, 32, 64, 128....) and each time we need to copy all elements to new place in memory. For really long strings like 1000 characters or more, your code will be 3 or more times slower then my soluton. Maybe try to make benchmark and show your solution on Q&A for points ofc :)